### ISSUE TITLE
Trying to use reusable block crashes the editor

### ISSUE THREAD

**Issue by IamLizu** [WordPress/gutenberg#17217]
Editor is crashing when I try to reuse a block from reusable blocks. The reusable block is mostly a custom html. I have 9-10 saved blocks, most of them are custom html, js to be specific.

**To reproduce**
1. Open editor on any post
2. First create a reusable block with custom html type
3. Update the post and exit editor
4. Reopen the editor on the same post or different, try using the created block
5. Editor will crash

**Expected behavior**
The editor is supposed add the saved block to the post.

**Screenshots**
![gutenbeg editor error](https://i.imgur.com/IE95ewg.jpg)

I am experiencing the issue on my computer, I don't know about smartphone. Here's the details,
 - OS: Windows 10
 - Browser: Chrome
 - Version 76.0.3809.100

My version of gutenberg came with WordPress version 5.2.2. I didn't do any manual installation or update.

**Comment by talldan** [comment]
@IamLizu Thanks for creating the issue.

I tried testing both in WP 5.2.2 and using the latest version of the Gutenberg code and wasn't able to reproduce this issue.

Is there an HTML snippet you can share that triggers the error, or are you just leaving the Custom HTML block blank?

**Comment by IamLizu** [comment]
Blocks are not blank. There's some images converted to base64 of which I often use in my post and pages, one twitter timeline embed.  Let me share some of the snippets, 

```
<div class="auth-c-container">
  <img src="/uploads/2019/07/imageOwnMin.png" alt="Avatar" style="width:90px">
  <p><span>S M Mahmudul Hasan</span> aka iamlizu</p>
  <p>A self taught guy, studies Software Engineering at American International University-Bangladesh(AIUB), creates technical content in spare times, mostly found of tech forums, terrible programmer, part-time gamer, listens to electronic musics and a huge nerd.</p>
</div>
```

```
<script type="text/javascript">
function disableSelection(e){if(typeof e.onselectstart!="undefined")e.onselectstart=function(){return false};else if(typeof e.style.MozUserSelect!="undefined")e.style.MozUserSelect="none";else e.onmousedown=function(){return false};e.style.cursor="default"}window.onload=function(){disableSelection(document.body)}
</script>
<script type="text/javascript">
document.oncontextmenu=function(e){var t=e||window.event;var n=t.target||t.srcElement;if(n.nodeName!="A")return false};
document.ondragstart=function(){return false};
</script>
<style type="text/css">
* : (input, textarea) {
	-webkit-touch-callout:none;
	-webkit-user-select:none;
}
</style>
<style type="text/css">
img {
	-webkit-touch-callout:none;
	-webkit-user-select:none;
}
</style>
<script type="text/javascript">
window.addEventListener("keydown",function(e){if(e.ctrlKey&&(e.which==65||e.which==66||e.which==67||e.which==70||e.which==73||e.which==80||e.which==83||e.which==85||e.which==86)){e.preventDefault()}});document.keypress=function(e){if(e.ctrlKey&&(e.which==65||e.which==66||e.which==70||e.which==67||e.which==73||e.which==80||e.which==83||e.which==85||e.which==86)){}return false};
window.addEventListener("keydown",function(e){if( event.metaKey&&(e.which==65||e.which==66||e.which==67||e.which==70||e.which==73||e.which==80||e.which==83||e.which==85||e.which==86)){e.preventDefault()}});document.keypress=function(e){if(e.ctrlKey&&(e.which==65||e.which==66||e.which==70||e.which==67||e.which==73||e.which==80||e.which==83||e.which==85||e.which==86)){}return false}
</script>
<script type="text/javascript">
document.onkeydown=function(e){e=e||window.event;if(e.keyCode==123||e.keyCode==18){return false}}
</script>
```

```
<script>
function init() {
var imgDefer = document.getElementsByTagName('img');
for (var i=0; i<imgDefer.length; i++) {
if(imgDefer[i].getAttribute('data-src')) {
imgDefer[i].setAttribute('src',imgDefer[i].getAttribute('data-src'));
} } }
window.onload = init;
</script>
```
There's one or two text blocks. And one block containing my social media pages. That's all.

**Comment by talldan** [comment]
The first one worked ok for me.

You won't be able to include scripts in a Custom HTML block.

**Comment by talldan** [comment]
There is a Twitter block you can use instead of pasting in the html for an embed.

**Comment by IamLizu** [comment]
> There is a Twitter block you can use instead of pasting in the html for an embed.

Yes, that twitter one is actually a saved twitter block.

**Comment by IamLizu** [comment]
> You won't be able to include scripts in a Custom HTML block.

Strange, I can add them and even save them, but I can't reuse them. I have added scripts into custom blocks in pages and posts, they are working. Am I missing something?

**Comment by IamLizu** [comment]
And let me clarify, the error happens every time whatever saved blocked i am trying to reuse.

**Comment by IamLizu** [comment]
Please advise me, should I remove the existing blocks and test with some new?

**Comment by talldan** [comment]
@IamLizu I was incorrect earlier, it looks like you can put scripts in the Custom HTML block.

I'm still not able to reproduce any errors, though, so I can't help there.

**Comment by IamLizu** [comment]
@talldan Sure, here's the error
```
TypeError: Cannot read property 'setContent' of null
    at t.value (https://iamlizu.com/incs/js/dist/block-library.min.js:23:41795)
    at qi (https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:130:296)
    at ui (https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:133:320)
    at https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:158:377
    at unstable_runWithPriority (https://iamlizu.com/incs/js/dist/vendor/react.min.js:27:36)
    at Vc (https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:158:360)
    at Sc (https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:158:144)
    at Z (https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:156:492)
    at Zg (https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:159:8)
    at Xe (https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:23:322)
```

**Comment by paaljoachim** [comment]
Btw: https://github.com/WordPress/gutenberg/pull/14367

**Comment by senadir** [comment]
I cannot reproduce this issue with of the snippets you posted

**Comment by IamLizu** [comment]
I have deleted all the reusable blocks permanently and added two new block. 
1. text block, a paragraph just says `Hello!`
2. A github gist embed that has nothing but `hello`.
   to be very clear, script contains, 
`<script src="https://gist.github.com/IamLizu/3f89b1ab2607a2d99f70f4fb10ce3e0d.js"></script>`  

Now when I try to use the added blocks (not both, just one) I get the same error screen as my first comment on this issue. The error contains,

```
TypeError: Cannot read property 'setContent' of null
    at t.value (https://iamlizu.com/incs/js/dist/block-library.min.js:23:41795)
    at qi (https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:130:296)
    at ui (https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:133:320)
    at https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:158:377
    at unstable_runWithPriority (https://iamlizu.com/incs/js/dist/vendor/react.min.js:27:36)
    at Vc (https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:158:360)
    at Sc (https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:158:144)
    at Z (https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:156:492)
    at Zg (https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:159:8)
    at Xe (https://iamlizu.com/incs/js/dist/vendor/react-dom.min.js:23:322)
```

> I cannot reproduce this issue with of the snippets you posted
@senadir I don't think its something to do with those snippets or the extra snippets I had. As I see it, I am getting same error these two new blocks.

**Comment by senadir** [comment]
I see that you're using the Gutenberg version that came with 5.2, that is Gutenberg 5.4
we're at version 6.4 & approaching 6.5 right now, this problem could have been fixed, and there were also new changes to reusable blocks that could have fixed this, I've included the latest build of Gutenberg, can you try it and tell if this persists?
[gutenberg.zip](https://github.com/WordPress/gutenberg/files/3597530/gutenberg.zip)

**Comment by IamLizu** [comment]
@senadir Yes, installing the plugin solved the issue. I have tested on multiple posts, editor is not crashing. I will have to use the plugin it seems, it could have been better if updates were included in the core.

**Comment by senadir** [comment]
Core updates are included with each major version, 5.3 will include 6.5, Gutenberg is released on a biweekly schedule, core is released on a 4-6 month cycle, so the plugin is the only way to stay on the bleeding edge
